exporter/datadogexporter/translate_traces.go

```diff
@@ -0,0 +1,492 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadogexporter
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"go.opencensus.io/trace"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+	tracetranslator "go.opentelemetry.io/collector/translator/trace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+)
+
+// codeDetails specifies information about a trace status code.
+type codeDetails struct {
+	message string // status message
+	status  int    // corresponding HTTP status code
+}
+
+const (
+	deploymentEnv       string = "deployment.environment"
+	keySamplingPriority string = "_sampling_priority_v1"
+	keySpanName         string = "span.name"
+	httpStatusCode      string = "http.status_code"
+	httpMethod          string = "http.method"
+	httpRoute           string = "http.route"
```

Comment:
These are available as http attribute names: https://github.com/open-telemetry/opentelemetry-collector/blob/c8622d2353485e8aa9ebde351feee431fc46f2f9/translator/conventions/opentelemetry.go#L183

Reply:
ah, nice, updated these to rely on convention constants when available.